x-pack/plugins/apm/public/components/alerting/alerting_flyout/index.tsx

```diff
@@ -30,9 +30,12 @@ export function AlertingFlyout(props: Props) {
 
   const serviceName = useServiceName();
   const { query } = useApmParams('/*');
-  const {
-    urlParams: { start, end },
-  } = useUrlParams();
+
+  const rangeFrom = 'rangeFrom' in query ? query.rangeFrom : undefined;
+  const rangeTo = 'rangeTo' in query ? query.rangeTo : undefined;
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo }, true);
```

Comment:
From out outside the last argument (`true`) is pretty opaque. Inspecting `useTimeRange` I can see it's to indicate whether `rangeFrom` and `rangeTo` are optional.

Reply:
This might be better

```suggestion
  const { start, end } = useTimeRange({ rangeFrom, rangeTo, optional: true });
```

Reply:
lgtm, let me see if I can make TypeScript happy with it

Reply:
Should be fixed now, thanks for the suggestion!